### PR TITLE
[LI-HOTFIX] Detect and automatically drop corrupted leader-epoch-checkpoint

### PR DESCRIPTION
**Pull request by gitlw**
TICKET = [LIKAFKA-47870](https://jira01.corp.linkedin.com:8443/browse/LIKAFKA-47870)
LI_DESCRIPTION =
In the ticket, we've identified the need to build a scanning mode such that the broker startup does not
need to be aborted. This PR is the 1st in a series to automatically detect and drop corrupted files or 
portions of corrupted files.

Specifically, this PR adds a flag `li.drop.corrupted.files.enable`. When the flag is enabled, the broker will
automatically clear the content of a corrupted leader-epoch-checkpoint file.
Dropping the leader-epoch-checkpoint file may possibly cause log divergence.
However, fixing such log divergence is out of the scope of this PR.

EXIT_CRITERIA = If and when this change is accepted in upstream kafka.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -327,6 +327,7 @@ object Defaults {
   val LiAsyncFetcherEnabled = false
   val LiNumControllerInitThreads = 1
   val LiLogCleanerFineGrainedLockEnabled = true
+  val LiDropCorruptedFilesEnabled = false
```
Comment (lmr3796) [inline core/src/main/scala/kafka/server/KafkaConfig.scala:330]: (Non-blocking) Do we want to separate the config for dropping log segments & checkpoint files?

File: core/src/main/scala/kafka/server/checkpoints/CheckpointFile.scala [hunk 2]
```diff
@@ -60,17 +59,27 @@ class CheckpointReadBuffer[T](location: String,
...
-          throw new IOException(s"Unrecognized version of the checkpoint file ($location): " + version)
+          maybeThrowException(new IOException(s"Unrecognized version of the checkpoint file ($location): " + line))
       }
     } catch {
-      case _: NumberFormatException => throw malformedLineException(line)
+      case _: NumberFormatException => maybeThrowException(malformedLineException(line))
+    }
+  }
+
+  private def maybeThrowException(e: Exception): Seq[T] = {
```
Comment (lmr3796) [inline core/src/main/scala/kafka/server/checkpoints/CheckpointFile.scala]: nit: The `maybeThrow` sounds a bit vague to me.  How about `dropCorruptFileOrThrow`

**Review by lmr3796 — CHANGES_REQUESTED** [review]

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 1 above)
Comment (gitlw) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:330]: For now, my thinking is that we don't need to have the separation.

File: core/src/main/scala/kafka/server/checkpoints/CheckpointFile.scala (hunk 2 above)
Comment (gitlw) [reply core/src/main/scala/kafka/server/checkpoints/CheckpointFile.scala]: Thanks for the suggestion. I've updated the method name.

**Review by lmr3796 — APPROVED** [review]

**Pull request merged**